src/Serial.cpp

```diff
@@ -52,9 +61,80 @@ namespace occa {
 #endif
     }
 
+    std::string getField(const std::string &shell_cmd,
+			 const std::string &field,
+			 bool ignoreCase,
+			 char delimiter){
+      // popen(), fgets(). strncasecmp() and isspace() are
+      // all available on Linux, BSD/OSX, CygWin and MinGW,
+      // assuming we're not compiling __STRICT_ANSI__
+#if (OCCA_OS & (LINUX_OS | WINUX_OS | OSX_OS))
+      FILE *fp;
+      fp = popen(shell_cmd.c_str(), "r");
+
+      const int bufferSize = 4096;
+      char *buffer = new char[bufferSize];
+
+      int field_len = field.size();
+
+      const char *field_str = field.c_str();
+      char *ptr;
+      char *str;
+
+      std::string ret = "";
+
+      while(fgets(buffer,bufferSize,fp)) {
```

Comment:
Sure, no problem. I did not mean to "disrespect" your style. FYI I like 
structured style too. Just about the only /occasional/ exception being 
(for me) when I do a "one-liner" while() such as when scanning strings 
like this. And sometimes "if (...) continue; / break;" in loops.  Just a 
habit for me nowadays. But you are correct, I must remain consistent 
with your style when making any mods.

On 05/02/2016 08:12 PM, David Medina wrote:
>
> In src/Serial.cpp 
> <https://github.com/libocca/occa/pull/30#discussion_r61829236>:
>
> > +#if (OCCA_OS & (LINUX_OS | WINUX_OS | OSX_OS))
> > +      FILE *fp;
> > +      fp = popen(shell_cmd.c_str(), "r");
> > +
> > +      const int bufferSize = 4096;
> > +      char *buffer = new char[bufferSize];
> > +
> > +      int field_len = field.size();
> > +
> > +      const char *field_str = field.c_str();
> > +      char *ptr;
> > +      char *str;
> > +
> > +      std::string ret = "";
> > +
> > +      while(fgets(buffer,bufferSize,fp)) {
>
> Can you put loop statements in new lines (89-107)
> For example
>
> while(*ptr && isspace(*ptr)) ++ptr;
>
> to
>
> while(*ptr !='\0'  && isspace(*ptr) {
>    ++ptr;
> }
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub 
> <https://github.com/libocca/occa/pull/30/files/67dfa18d76aaaf3c49d3dad2bac25f051d523320#r61829236>
>